src/Valet/Services/ConfigurationService.cs

```diff
@@ -0,0 +1,50 @@
+using System.Text;
+using Sharprompt;
+using Valet.Interfaces;
+
+namespace Valet.Services;
+
+public class ConfigurationService : IConfigurationService
+{
+    public async Task<Dictionary<string, string>> ReadCurrentVariablesAsync(string filePath = ".env.local")
+    {
+        var lines = await File.ReadAllLinesAsync(filePath);
+
+        var variables = new Dictionary<string, string>();
+        foreach (var line in lines)
+        {
+            if (string.IsNullOrWhiteSpace(line)) continue;
+
+            var variable = line.Split('=', StringSplitOptions.TrimEntries);
+            if (variable.Length != 2 || string.IsNullOrWhiteSpace(variable[1])) continue;
+
+            variables[variable[0]] = variable[1];
+        }
+
+        return variables;
+    }
+
+    public Dictionary<string, string> GetUserInput()
+    {
+        var input = new Dictionary<string, string>();
+
+        foreach (var variable in Constants.UserInputVariables)
```

Comment:
💭 What do you think of adding another parameter to configure: `gh valet configure azure-devops`

I'm thinking most customers will only want to use valet with a single provider, and adding all the other env variables might be a bit overkill. Plus, this would have the side benefit of highlighting exactly what they would need to use valet for each provider

Reply:
Definitely like that in concept! We've seen a non-trivial amount of customers using more than one provider, so maybe the first step is presenting a multi-select prompt for a user to select one or many providers they are looking to configure. Then, we would only prompt for the values that are relevant..

<img width="1450" alt="Screen Shot 2022-06-08 at 10 14 03 AM" src="https://user-images.githubusercontent.com/8703324/172677178-5a655e6d-9c04-49eb-9d65-421856255a68.png">

Reply:
That looks sweet!

Reply:
This has been implemented in https://github.com/github/gh-valet/pull/45/commits/5ffbdc7a7fa728edb91484064e739fcb641480f4